test/androidTest/java/org/thoughtcrime/securesms/espresso/ConversationHelper.java

```diff
@@ -89,6 +93,36 @@ public ConversationHelper sendReaction(int position, int emoji) {
         return this;
     }
 
+    public ConversationHelper sendReply(int position, String message) {
+        this.unselectMessage();
+
+        this.selectMessage(position);
+
+        onView(withId(R.id.menu_context_reply))
+            .perform(click());
+        onView(withId(R.id.lin_reply))
+            .perform(click());
+        onView(withId(R.id.custom_reply))
+            .perform(typeText(message))
+
+            .perform(android.support.test.espresso.action.ViewActions.pressKey(KeyEvent.KEYCODE_ENTER))
+            .perform(replaceText(""))
+            .perform(android.support.test.espresso.action.ViewActions.pressKey(KeyEvent.KEYCODE_BACK))
```

Comment:
What are these last two lines doing?

Reply:
1st back hides keyboard, 2nd back exits reply mode. Could not find a way to exit search mode besides that